File: src/main/java/com/elchologamer/userlogin/database/sql/PostgreSQL.java
package com.elchologamer.userlogin.database.sql;

import org.bukkit.configuration.ConfigurationSection;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;
import java.util.Properties;

public class PostgreSQL extends SQLDatabase {

    public PostgreSQL() {
        super("postgresql", "org.postgresql.Driver");
    }

    @Override
    public void connect() throws SQLException, ClassNotFoundException {
        super.connect();

        // Update password max length
        int maxChars = Math.max(
                getPlugin().getConfig().getInt("password.maxCharacters", 128),
                1
        );

        update(
                "ALTER TABLE " + getTable() + " " +
                        "ALTER COLUMN password " +
                        "TYPE VARCHAR(" + maxChars + ")"
        );
    }

    @Override
    protected Connection getConnection() throws SQLException {
        ConfigurationSection section = getPlugin().getConfig()
                .getConfigurationSection("database.postgresql");
        assert section != null;

        String host = section.getString("host", "localhost");
        int port = section.getInt("port", 5432);
        boolean ssl = section.getBoolean("ssl");

        String username = section.getString("username", "root");
        String password = section.getString("password", null);

        Properties props = new Properties();

        props.setProperty("user", username);
        if (password != null) props.setProperty("password", password);
        props.setProperty("ssl", Boolean.toString(ssl));

        String url = "jdbc:postgresql://" + host + ":" + port + "/" + getDatabase();
        return DriverManager.getConnection(url, props);
    }
}

File: src/main/kotlin/com/elchologamer/userlogin/database/sql/PostgreSQL.kt
package com.elchologamer.userlogin.database.sql

import com.elchologamer.userlogin.UserLogin.Companion.plugin
import org.bukkit.configuration.ConfigurationSection
import java.sql.Connection
import java.sql.DriverManager
import java.util.*

class PostgreSQL : SQLDatabase("postgresql", "org.postgresql.Driver") {
    override fun connect() {
        super.connect()

        // Update password max length
        val maxChars = plugin.config.getInt("password.maxCharacters", 128).coerceAtLeast(1)

        update(
            "ALTER TABLE $table " +
                    "ALTER COLUMN password " +
                    "TYPE VARCHAR($maxChars)"
        )
    }

    override fun getConnection(): Connection {
        val section: ConfigurationSection = plugin.config.getConfigurationSection("database.postgresql")
            ?: throw IllegalStateException("PostgreSQL settings missing in config.yml")

        val host = section.getString("host", "localhost")
        val port = section.getInt("port", 5432)
        val ssl = section.getBoolean("ssl")
        val username = section.getString("username", "root")
        val password = section.getString("password", null)

        val props = Properties()

        props.setProperty("user", username)
        password?.let { props.setProperty("password", it) }
        props.setProperty("ssl", ssl.toString())

        return DriverManager.getConnection("jdbc:postgresql://$host:$port/$database", props)
    }
}